std/datetime/systime.d

```diff
@@ -8751,26 +8751,27 @@ public:
         auto found = (skipFirst ? str[1..$] : str).byCodeUnit.find('.', 'Z', '+', '-');
         auto dateTimeStr = str[0 .. $ - found[0].length];
 
+        typeof(str.byCodeUnit) foundTZ; // needs to have longer lifetime than zoneStr
         typeof(str) fracSecStr;
         typeof(str) zoneStr;
 
         if (found[1] != 0)
         {
             if (found[1] == 1)
             {
-                auto foundTZ = found[0].find('Z', '+', '-');
+                foundTZ = found[0].find('Z', '+', '-')[0];
```

Comment:
I can't assign a Tuple here, since `Tuple.opAssign` is not `scope` compatible, so I keep the string and discard the index of the matched string. It's only used in a `!= 0` comparison to see if there was any match, so I replaced it by a `foundTZ.length` check.